### ISSUE TITLE
'disabled' attribute value cannot be removed on node with phx-update="ignore"

### ISSUE THREAD

**Issue by ftes** [phoenixframework/phoenix_live_view#3044]
### Environment

* Elixir version (elixir -v): Erlang/OTP 26 [erts-14.2.1] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit:ns]
* Phoenix version (mix deps): 1.17.10
* Phoenix LiveView version (mix deps): 0.20.3
* Operating system: macOS
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome, Firefox, Safari
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

<details>
<summary>Single file app</summary>

```ex
Application.put_env(:sample, Example.Endpoint,
  http: [ip: {127, 0, 0, 1}, port: 5001],
  server: true,
  live_view: [signing_salt: "aaaaaaaa"],
  secret_key_base: String.duplicate("a", 64)
)

Mix.install([
  {:plug_cowboy, "~> 2.5"},
  {:jason, "~> 1.0"},
  {:phoenix, "~> 1.7.10"},
  {:phoenix_live_view, "~> 0.20.3"}
])

defmodule Example.ErrorView do
  def render(template, _), do: Phoenix.Controller.status_message_from_template(template)
end

defmodule Example.BugLive do
  use Phoenix.LiveView, layout: {__MODULE__, :live}

  def mount(_, _, socket) do
    {:ok, assign(socket, :disabled, false)}
  end

  defp phx_vsn, do: Application.spec(:phoenix, :vsn)
  defp lv_vsn, do: Application.spec(:phoenix_live_view, :vsn)

  def render("live.html", assigns) do
    ~H"""
    <script src={"https://cdn.jsdelivr.net/npm/phoenix@#{phx_vsn()}/priv/static/phoenix.min.js"}>
    </script>
    <script
      src={"https://cdn.jsdelivr.net/npm/phoenix_live_view@#{lv_vsn()}/priv/static/phoenix_live_view.min.js"}
    >
    </script>
    <script>
      let liveSocket = new window.LiveView.LiveSocket("/live", window.Phoenix.Socket)
      liveSocket.connect()
    </script>
    <style>
      * { font-size: 1.1em; }
    </style>
    <%= @inner_content %>
    """
  end

  def render(assigns) do
    ~H"""
    <input
      value={if @disabled, do: "disabled", else: "not disabled"}
      disabled={@disabled}
      phx-update="ignore"
      id="test-input"
    />
    <button phx-click="toggle">Toggle</button>
    """
  end

  def handle_event("toggle", _, socket) do
    {:noreply, update(socket, :disabled, &(not &1))}
  end
end

defmodule Example.Router do
  use Phoenix.Router
  import Phoenix.LiveView.Router

  pipeline :browser do
    plug(:accepts, ["html"])
  end

  scope "/", Example do
    pipe_through(:browser)

    live("/", BugLive, :index)
  end
end

defmodule Example.Endpoint do
  use Phoenix.Endpoint, otp_app: :sample
  socket("/live", Phoenix.LiveView.Socket)
  plug(Example.Router)
end

{:ok, _} = Supervisor.start_link([Example.Endpoint], strategy: :one_for_one)
Process.sleep(:infinity)
```

</details>

For an `<input phx-update="ignore" />` node:
- The `value` attribute of the input node is updated on every button click ✅ (text in input changes).
- The `disabled` attribute of the input node is added on the first button click, but not removed on the second click ❌ (input stays "greyed out").

https://github.com/phoenixframework/phoenix_live_view/assets/2614732/f6f2e6cc-fe81-4c5d-ac16-1ce189da76e4

```ex
  def render(assigns) do
    ~H"""
    <input
      value={if @disabled, do: "disabled", else: "not disabled"}
      disabled={@disabled}
      phx-update="ignore"
      id="test-input"
    />
    <button phx-click="toggle">Toggle</button>
    """
  end

  def handle_event("toggle", _, socket) do
    {:noreply, update(socket, :disabled, &(not &1))}
  end
```

I also added tests for comparison: https://github.com/ftes/phx-update-ignore-attr-disabled-not-updated
- [LiveView test](https://github.com/ftes/phx-update-ignore-attr-disabled-not-updated/blob/master/test/disabled_bug_web/disabled_bug_live_test.exs) passes (so the test DOM implementation works as I would expect)
- [Wallaby test](https://github.com/ftes/phx-update-ignore-attr-disabled-not-updated/blob/master/test/disabled_bug_web/disabled_bug_live_feature_test.exs) fails (so the actual behaviour in a browser is broken, as you can see in the screencast or when trying it yourself)

### Expected behavior

The `disabled` attribute of the input node is removed on the second button click.
In other words: after clicking the button a second time, the input should not be disabled (aka should be enabled).

From [the docs](https://hexdocs.pm/phoenix_live_view/dom-patching.html):
> [For phx-update="ignore"] note only the element contents are ignored, its attributes can still be updated.

**Comment by SteffenDE** [comment]
Can you try `{:phoenix_live_view, github: "phoenixframework/phoenix_live_view", branch: "main", override: true}`?

**Comment by SteffenDE** [comment]
Can reproduce on main.

**Comment by SteffenDE** [comment]
Because value is only changed and not toggled. This only affects parameters that are removed. Basically currently you can only add or change parameters on phx-update ignore, except if they start with `data-`.

**Comment by ftes** [comment]
Thank you. I have slightly altered the issue title and description to be more accurate.
Since I didn't distinguish between updating and toggling an attribute.

**Comment by chrismccord** [comment]
We only merge attributes from the server, but we cannot remove them as the client may be setting attributes itself on the ignored container. For example if you never rendered the value on the server, and instead set it on the client,the patch from #3045 would remove the value. In this case, you can wrap the input in its own phx-update=ignore container which should handle your case. Thanks!

**Comment by ftes** [comment]
Thanks @chrismccord for your feedback!
While I understand the reasoning, I would like to point out three inconsistencies and make suggestions how to address them:

1. Tests: DOM patching in Liveview tests does not match actual JS DOM patching as I showed in these side-by-side tests: [LiveView test](https://github.com/ftes/phx-update-ignore-attr-disabled-not-updated/blob/master/test/disabled_bug_web/disabled_bug_live_test.exs), [Wallaby test](https://github.com/ftes/phx-update-ignore-attr-disabled-not-updated/blob/master/test/disabled_bug_web/disabled_bug_live_feature_test.exs)
2. "Special" attributes: `data-*` attributes _can_ be removed on the ignored container. Could you elaborate why your remark should apply to other attributes, but not `data-` ones?
>  we cannot remove them as the client may be setting attributes itself on the ignored container.
3. Removing vs. adding vs. updating: The server is allowed to add and update attributes on ignored containers, but not remove them.
  a. Why not apply your reasoning to adding and updating as well? "we cannot [add or overwrite] them as the client may be [overwriting values or removing attributes] itself on the ignored container"
  b. What makes the "removing" case so special? Isn't it a general question of who is in charge - the client or the server?
    1. Full client control: DOM node nested inside an ignored container
    2. Full server control: outside of ignored container
    3. Mixed control: attributes on an ignored container. Here the Liveview user has to be aware of the implications. But I don't see why adding and updating should be privileged operations the server is allowed to perform.

At the least, I think the existing inconsistencies should be documented. I'm happy to create a PR if you agree.

Ideally, imho things should be more consistent. Consistent is easier to understand. Debugging this problem was especially hard because `data-disabled` was toggling fine, while `disabled` was not. And because Liveview tests were behaving different from reality.

I do understand that removing inconsistencies would result in a breaking change. Because as Hyrum's law tells us, users have likely come to depend on these hidden inconsistencies.

**Comment by ftes** [comment]
@SteffenDE Do you see think it is valuable to pursue any of the points I made above? ☝️

**Comment by SteffenDE** [comment]
@ftes yes, definitely. I think these are some good points to discuss. There is this commit that adjusted the documentation a little bit: https://github.com/phoenixframework/phoenix_live_view/commit/a774138564f01425c43241f51fb4ae77f85c5726

LiveViewTest should handle this the same way, I created https://github.com/phoenixframework/phoenix_live_view/issues/3119 to track this.

Considering points 2 or 3, I think that's something for Chris to elaborate on.

**Comment by josevalim** [comment]
Correct, we did make it more consistent in a follow up commit and documented it, exactly to address @ftes concerns, but we forgot to follow up:  https://github.com/phoenixframework/phoenix_live_view/commit/a774138564f01425c43241f51fb4ae77f85c5726

That commit and docs should be taken as the source of truth.

**Comment by ftes** [comment]
Thank you very much for following up on this!
